nexus/db-queries/src/db/datastore/v2p_mapping.rs

```diff
@@ -0,0 +1,51 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use super::DataStore;
+use crate::context::OpContext;
+use crate::db;
+use crate::db::datastore::SQL_BATCH_SIZE;
+use crate::db::error::{public_error_from_diesel, ErrorHandler};
+use crate::db::model::V2PMappingView;
+use crate::db::pagination::paginated;
+use crate::db::pagination::Paginator;
+use async_bb8_diesel::AsyncRunQueryDsl;
+use diesel::{QueryDsl, SelectableHelper};
+use omicron_common::api::external::ListResultVec;
+
+impl DataStore {
+    pub async fn v2p_mappings(
+        &self,
+        opctx: &OpContext,
+    ) -> ListResultVec<V2PMappingView> {
+        use db::schema::v2p_mapping_view::dsl;
+
+        opctx.check_complex_operations_allowed()?;
+
+        let mut mappings = Vec::new();
+        let mut paginator = Paginator::new(SQL_BATCH_SIZE);
+        while let Some(p) = paginator.next() {
+            let batch = paginated(
+                dsl::v2p_mapping_view,
+                dsl::sled_id,
+                &p.current_pagparams(),
+            )
+            .select(V2PMappingView::as_select())
+            .load_async(&*self.pool_connection_authorized(opctx).await?)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+            paginator = p.found_batch(&batch, &|mapping| mapping.nic_id);
```

Comment:
`mapping.nic_id` here is telling the paginator to fetch the next page based on NIC ID, but the `paginated` call above is using `dsl::sled_id`. I think these need to match - it should be possible to craft a test that fails with the code as written, I think? (Or maybe the current tests would fail after removing the non-paginated query below that @FelixMcFelix pointed out?)

Reply:
I _think_ if we moved these columns to typed UUIDs, this would fail to compile entirely - @sunshowers do you know if `DbTypedUuid` can be used through `paginated`?

Reply:
@jgallagher @sunshowers if it's possible to used typed UUIDs to guard against this without too much rework I'd like to do so :D